src/app/notes/notes.service.ts

```diff
@@ -19,7 +20,22 @@ export class NotesService {
    * @returns The NoteList as observable
    */
   getNotes(): Observable<NoteList> {
-    this.logger.debug(`Gathering notes from ${environment.notesAsset}`);
-    return this.http.get<NoteList>(environment.notesAsset).pipe();
+    this.logger.debug(`Gathering notes from ${environment.notesAssets}`);
+
+    const observables = [];
+    for (const jsonFile of environment.notesAssets) {
+      observables.push(this.http.get<NoteList>(jsonFile));
+    }
+
+    return forkJoin(observables).pipe(
+      map((notes: Array<NoteList>) => {
+        if (notes.length > 1) {
+          for (const key of Object.keys(notes[1])) {
```

Comment:
@Drogglbecher The mentioned fix does not work for me. If I insert something like this 
```suggestion
          for (const key of Object.keys(notes[1])) {
                    console.log(notes[0].get(0));
```
Then this results in another runtime type error (but compiles):
```
TypeError: notes[0].get is not a function
```

Reply:
So to me it looks like we're tricking the compiler here. It is always an `any` object but we say that is a `NoteList`. So the type is simply not correct. :man_shrugging:

Reply:
I'm a little bit confused since I'm not sure whether the question above was another context.

`new Map([...notes[0], ...notes[1]])` creates a merged map of the both maps and the result should be accessed with `get`. When you have an `Array` of `Map`s, your syntax seems to be right.

Since you load data async of an unknown source, the typechecker could obviously not ensure type safety while run time, so you are absolutely right. When there occurs some kind of weird type, which is not the one you typed, it crashes. You can check it with `typeof` or just debug and have a look what get's returned there.

Reply:
Okay, thanks. :pray: But the angular `HttpClient` provides a generic `get`, which we use here to assert the correct type. The request did not fail, but the type during runtime is not the correct one. I'm not sure if Angular needs to do further processing here, but I opened up a bug to clarify the expected behavior: https://github.com/angular/angular/issues/31610